### PR TITLE
feat: add `share` operator

### PR DESCRIPTION
**Pull request by M-Adoo**

### CODE REVIEW

**Comment by Mnwa** [comment]
@Mergifyio rebase

File: src/observable.rs [hunk 1]
```diff
@@ -782,6 +783,24 @@ pub trait Observable {
     }
   }
 
+  /// Returns a new Observable that multicast (shares) the original
+  /// Observable. As long as there is at least one Subscriber this
+  /// Observable will be subscribed and emitting data. When all subscribers
+  /// have unsubscribed it will unsubscribe from the source Observable.
+  /// Because the Observable is multicasting it makes the stream `hot`.
+  /// This is an alias for `publish().refCount()`
```
Comment (Mnwa) [inline src/observable.rs]: ref_count

Comment (M-Adoo) [reply src/observable.rs]: ok, I'll fix it tomorrow.

**Comment by M-Adoo** [comment]
Rebase may cause confusion for contributors,  we should use this command before we decide merge this pr.

File: src/observable.rs (hunk 1 above)
Comment (M-Adoo) [reply src/observable.rs]: I fixed it now, I choose to fix it before going to bed.  Please review it again.

**Pull request merged**